tower/src/util/combinators/with.rs

```diff
@@ -0,0 +1,63 @@
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+use tower_layer::Layer;
+use tower_service::Service;
+
+#[derive(Debug)]
+pub struct With<S, F> {
```

Comment:
...and here (even though CI isn't warning about it for some reason? are the types in this file not being exported publicly?)